lib/kafkaClient.js

```diff
@@ -1125,6 +1125,30 @@ KafkaClient.prototype.sendFetchRequest = function (
   maxTickMessages,
   callback
 ) {
+  payloads = _.cloneDeep(payloads);
+  const memberId = consumer.memberId;
+  const generationId = consumer.generationId;
+  function stateValidator (unused, type, message) {
+    const payloadMap = consumer.payloadMap;
+    if (consumer.closing || consumer.connecting || consumer.rebalancing || consumer.memberId !== memberId || consumer.generationId !== generationId) {
+      logger.error('ignoring message due to it being from an old group - memberId: ' + memberId, '!=' + consumer.memberId + ' - generationId: ' + generationId + '!=' + consumer.generationId);
+      return false;
+    }
+    if (type === 'message') {
+      const {topic, partition, offset} = message;
+      if (!payloadMap[topic] || payloadMap[topic][partition] == null) {
+        // We should have never received this in the first place
```

Comment:
can we log an error for this case?

Reply:
good idea, since old memberships should be filtered from above